pandas/core/indexing.py

```diff
@@ -1481,7 +1481,10 @@ def _validate_key(self, key, axis: AxisInt):
             # so don't treat a tuple as a valid indexer
             raise IndexingError("Too many indexers")
         elif is_list_like_indexer(key):
-            arr = np.array(key)
+            if isinstance(key, ABCSeries):
+                arr = key._values
```

Comment:
How about integer arrays with `pd.NA`?

Reply:
Is validated later

Reply:
The errors are different for Series and Array:

```python
>>> import pandas as pd
>>> df = pd.DataFrame([[0,1,2,3,4],[5,6,7,8,9]])
>>> iarr = pd.array([0,1,2, pd.NA], dtype = pd.Int64Dtype())
>>> df.iloc[:, iarr]
IndexError: .iloc requires numeric indexers, got [0 1 2 <NA>]
df.iloc[:, pd.Series(iarr)]
ValueError: cannot convert to 'int64'-dtype NumPy array with missing values. Specify an appropriate 'na_value' for this dtype.
```
I think the ValueError should be raised in both cases?

Reply:
I guess you tried the series case on main? I am getting consistent errors on my pr. 

In General I would prefer a better suited error, pointing to NA in iloc. But will probably do in a follow up. We will have to check this later, because not all cases get here and this does not cover the series.iloc case

Reply:
Sorry wrong snippet. It's the non-NA array:

>>> import pandas as pd
>>> df = pd.DataFrame([[0,1,2,3,4],[5,6,7,8,9]])
>>> iarr = pd.array([0,1,2], dtype = pd.Int64Dtype())
>>> df.iloc[:, iarr]
IndexError: .iloc requires numeric indexers, got [0 1]
```

Reply:
Ah got you, fixed that as well